### PR TITLE
8328386: Convert java/awt/FileDialog/FileNameOverrideTest test to main

### PR DESCRIPTION
**Pull request by azuev-java**
Convert applet based test to a main based test;
Move test to an apporpriate location;
Remove old files;

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328386](https://bugs.openjdk.org/browse/JDK-8328386): Convert java/awt/FileDialog/FileNameOverrideTest test to main (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to [5a52b8ef](https://git.openjdk.org/jdk/pull/18365/files/5a52b8ef32741823ea50cb713e76fadf046eb07b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18365/head:pull/18365` \
`$ git checkout pull/18365`

Update a local copy of the PR: \
`$ git checkout pull/18365` \
`$ git pull https://git.openjdk.org/jdk.git pull/18365/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18365`

View PR using the GUI difftool: \
`$ git pr show -t 18365`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18365.diff">https://git.openjdk.org/jdk/pull/18365.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18365#issuecomment-2007300946)

### CODE REVIEW

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        frame.pack();
+        return frame;
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        String instructions = """
+                1) Click on 'Show File Dialog' button. A file dialog will come up.
+                2) Double-click on '$clickDirName' and click OK.
+                3) The text in the text field will indicate if test is passed or failed.
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:93]: It looks like we can remove the text field and its output altogether, and use `PassFailJFrame#forcePass`, `PassFailJFrame#forceFail` to finish the test.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    The test is not applicable for $toolkit. Press Pass.
+                    """. replace("$toolkit", toolkit);
+        }
+
+        PassFailJFrame.builder()
+                .title("File Dialog Return Test Instructions")
+                .instructions(instructions)
+                .rows((int) instructions.lines().count() + 1)
+                .columns(50)
+                .testUI(FileNameOverrideTest::initialize)
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:108]: It is possible to get rid of the extra frame, e.g. make the `initialize` return the "Show File Dialog" `JButton`

```suggestion
                .splitUIBottom(FileNameOverrideTest::initialize)
```

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+/*
+ * @test
+ * @bug 6260659
+ * @summary File Name set programmatically in FileDialog is overridden during navigation, XToolkit
+ * @requires (os.family == "linux")
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FileNameOverrideTest
+ */
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:32]: We usually move this block just before a class declaration, for better readability.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                1) Click on 'Show File Dialog' button. A file dialog will come up.
+                2) Double-click on '$clickDirName' and click OK.
+                3) The text in the text field will indicate if test is passed or failed.
+                """;
+
+        String toolkit = Toolkit.getDefaultToolkit().getClass().getName();
+        if (!toolkit.equals("sun.awt.X11.XToolkit")) {
+            instructions = """
+                    The test is not applicable for $toolkit. Press Pass.
+                    """. replace("$toolkit", toolkit);
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:100]: There is no point in showing the instruction window in this case, it is probably better to just throw the `jtreg.SkippedException`.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        return frame;
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        String instructions = """
+                1) Click on 'Show File Dialog' button. A file dialog will come up.
+                2) Double-click on '$clickDirName' and click OK.
+                3) The text in the text field will indicate if test is passed or failed.
+                """;
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:94]:
```suggestion
                """.replace("$clickDirName", clickDirName);
```

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java (hunk 1 above)
Comment (azuev-java) [reply test/jdk/java/awt/FileDialog/FileNameOverrideTest.java]: Fixed.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            Frame frame = new Frame();
+            fd = new FileDialog(frame, "Open");
+            fd.setFile(fileName);
+            fd.setDirectory(dirPath);
+            fd.setVisible(true);
+            String output = fd.getFile();
+            fd.dispose();
+            frame.dispose();
+            if (fileName.equals(output)) {
+                PassFailJFrame.forcePass();;
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:64]:
```suggestion
                PassFailJFrame.forcePass();
```

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    private final static String clickDirName = "Directory for double click";
+    private final static String dirPath = System.getProperty("user.dir");;
+    private static JButton showBtn;
+    private static FileDialog fd;
+
+    private static JButton initialize() {
+        showBtn = new JButton("Show File Dialog");
+        showBtn.addActionListener(w -> {
+            Frame frame = new Frame();
+            fd = new FileDialog(frame, "Open");
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:56]:
```suggestion
            fd = new FileDialog((Frame) null, "Open");
```
There is no point in this frame.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
+ * @run main/manual FileNameOverrideTest
+ */
+
+public class FileNameOverrideTest {
+    private final static String fileName = "input";
+    private final static String clickDirName = "Directory for double click";
+    private final static String dirPath = System.getProperty("user.dir");;
+    private static JButton showBtn;
```
Comment (azvegint) [inline test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:49]: Can be local variable

**Review by azvegint — APPROVED** [review]

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java (hunk 7 above)
Comment (azuev-java) [reply test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:56]: Ok.

File: test/jdk/java/awt/FileDialog/FileNameOverrideTest.java (hunk 8 above)
Comment (azuev-java) [reply test/jdk/java/awt/FileDialog/FileNameOverrideTest.java:49]: Fixed.

**Pull request closed without merging**